### PR TITLE
Add Necessary encodings for Local, Add ExecutionMode to the Evaluate API

### PR DESCRIPTION
**Pull request by jmcardon**
Changes in this PR are purely for chainweb's API and thus do not impact pact's test suite.

Test Coverage is done on chainweb directly.

PR checklist:

* [x] Test coverage for the proposed changes
* [x] PR description contains example output from repl interaction or a snippet from unit test output
* [x] (If Relevant) Documentation has been (manually) updated at https://docs.kadena.io/pact

Additionally, please justify why you should or should not do the following:

* [ ] Benchmark regressions
* [x] Confirm replay/back compat (Ignore until core release)
* [x] (For Kadena engineers) Run integration-tests against a Chainweb built with this version of Pact (Ignore until core release)

### CODE REVIEW

File: pact-request-api/Pact/Core/Command/Types.hs [hunk 1]
```diff
@@ -145,16 +145,13 @@ type SigCapability = CapToken QualifiedName PactValue
 -- | Pair parsed Pact expressions with the original text.
 data ParsedCode = ParsedCode
   { _pcCode :: !Text
-  , _pcExps :: ![Lisp.TopLevel ()]
+  , _pcExps :: ![Lisp.TopLevel Info]
```
Comment (rsoeldner) [inline pact-request-api/Pact/Core/Command/Types.hs:148]: Why this change to `SpanInfo`?

Comment (jmcardon) [reply pact-request-api/Pact/Core/Command/Types.hs:148]: Good question. 

We want chainweb's choice of `Info` to be governed by the `Info` alias in `Evaluate.hs`

`ParsedCode` was nuking line information before. Now we get to keep it and be consistent.

**Review by rsoeldner — APPROVED** [review]
